storage_controller/src/scheduler.rs

```diff
@@ -843,7 +845,7 @@ impl Scheduler {
         // Find the AZ with the lowest number of shards currently allocated
         Some(
             azs.into_iter()
-                .min_by_key(|i| (i.1.home_shard_count, i.0))
+                .min_by_key(|i| (i.1.home_shard_count / i.1.node_count, i.0))
```

Comment:
I wonder about rounding issues here: if there is 10 nodes in an AZ, there is no distinction between there being 8 shards vs 1 shard. Should we multiply the shard count by the least common multiple of all node counts maybe?

Alternatively we can use floats but those are cursed :laughing:

Reply:
I think we can assume that shard_count >> node_count.

For cases where they are roughly equal, each shard basically has a node to itself, so we don't particularly care which AZ it ends up in I think.

Reply:
Ahhh sorry, yeah, we are looking at total per-node shard numbers, not at shard per-tenant shard numbers. So that inequality does hold, you're right.